packages/error-reporting/src/configuration.js

```diff
@@ -0,0 +1,449 @@
+ * set the project id this function will defer next to the value set in the
+ * environment named `GCLOUD_PROJECT` if it is set and of type string. If this
+ * environmental variable is not set the function will defer to the
+ * _givenConfiguration property if it is of type object and has a string
+ * property named projectId. If none of these conditions are met then the
+ * _projectId property will be left at its default value.
+ * @memberof Configuration
+ * @private
+ * @function _checkLocalProjectId
+ * @param {Function} cb - The original user callback to invoke with the project
+ *  id or error encountered during id capture
+ * @returns {Undefined} - does not return anything
+ */
+Configuration.prototype._checkLocalProjectId = function(cb) {
+  if (isString(this._projectId)) {
+    // already has been set by the metadata service
+    return this._projectId;
+  } else if (has(this._givenConfiguration, 'projectId')) {
+    if (isString(this._givenConfiguration.projectId)) {
+      this._projectId = this._givenConfiguration.projectId;
+    } else if (isNumber(this._givenConfiguration.projectId)) {
+      this._projectId = this._givenConfiguration.projectId.toString();
+    }
+  }
+  return this._projectId;
+};
+/**
+ * Returns the _reportUncaughtExceptions property on the instance.
+ * @memberof Configuration
+ * @public
+ * @function getReportUncaughtExceptions
+ * @returns {Boolean} - returns the _reportUncaughtExceptions property
+ */
+Configuration.prototype.getReportUncaughtExceptions = function() {
+  return this._reportUncaughtExceptions;
+};
+/**
+ * Returns the _shouldReportErrorsToAPI property on the instance.
+ * @memberof Configuration
+ * @public
+ * @function getShouldReportErrorsToAPI
+ * @returns {Boolean} - returns the _shouldReportErrorsToAPI property
+ */
+Configuration.prototype.getShouldReportErrorsToAPI = function() {
+  return this._shouldReportErrorsToAPI;
+};
+/**
+ * Returns the _projectId property on the instance.
+ * @memberof Configuration
+ * @public
+ * @function getProjectId
+ * @returns {String|Null} - returns the _projectId property
+ */
+Configuration.prototype.getProjectId = function(cb) {
+  return this._checkLocalProjectId();
+};
+/**
+ * Returns the _key property on the instance.
+ * @memberof Configuration
+ * @public
+ * @function getKey
+ * @returns {String|Null} - returns the _key property
+ */
+Configuration.prototype.getKey = function() {
+  return this._key;
+};
+/**
+ * Returns the _keyFilename property on the instance.
+ * @memberof Configuration
+ * @public
+ * @function getKeyFilename
+ * @returns {String|Null} - returns the _keyFilename property
+ */
+Configuration.prototype.getKeyFilename = function() {
+  return this._keyFilename;
+};
+/**
+ * Returns the _credentials property on the instance.
+ * @memberof Configuration
+ * @public
+ * @function getCredentials
+ * @returns {Credentials\Null} - returns the _credentials property
```

Comment:
I think technically that's *more* correct, but I didn't realize that this files docs probably won't even make it to the site until after I commented. This specific syntax just throws a parse error in dox.